resources/views/platform/components/game/related-games-table.blade.php

```diff
@@ -0,0 +1,187 @@
+                        <td class='text-right'>{!! localized_number($game['points_total']) !!}</td>
+                        <td class='text-right'>{!! sprintf("%01.2f", $game['RetroRatio']) !!}</td>
+                    @endif
+
+                    @if ($game['leaderboards_count'] == 0)
+                        <td></td>
+                    @else
+                        <td class='text-right'>{!! localized_number($game['leaderboards_count']) !!}</td>
+                    @endif
+
+                    @if ($game['players_total'] == 0)
+                        <td></td>
+                    @else
+                        <td class='text-right'>{!! localized_number($game['players_total']) !!}</td>
+                    @endif
+
+                    @if ($game['NumTickets'] == 0)
+                        <td></td>
+                    @else
+                        <td class='text-right'><a href="/ticketmanager.php?g={{ $game['ID'] }}">{!! localized_number($game['NumTickets']) !!}</a></td>
+                    @endif
+
+                    @if ($userProgress !== null)
+                        @if ($game['achievements_published'] == 0)
+                            <td></td>
+                        @else
+                            <td>
+                            <?php
+                                $hardcoreProgressBarWidth = $softcoreProgressBarWidth = 0;
+                                $gameProgress = $userProgress[$game['ID']] ?? null;
+                                $achievementsUnlocked = 0;
+                                if ($gameProgress != null && $game['achievements_published']) {
+                                    $achievementsUnlocked = $gameProgress['achievements_unlocked'];
+                                    $hardcoreProgressBarWidth = sprintf("%01.2f", $gameProgress['achievements_unlocked_hardcore'] * 100 / $game['achievements_published']);
+                                    $softcoreProgressBarWidth = sprintf("%01.2f", ($achievementsUnlocked - $gameProgress['achievements_unlocked_hardcore']) * 100 / $game['achievements_published']);
+                                }
+                            ?>
+                            <div role="progressbar" aria-valuemin="0" aria-valuemax="100"
+                                title="{{ $achievementsUnlocked }} of {{ $game['achievements_published'] }} unlocked"
+                                class="w-full h-1 bg-embed rounded flex">
+                                <div style="width: {{ $hardcoreProgressBarWidth }}%"
+                                        class="bg-[#cc9900] h-full {{ $hardcoreProgressBarWidth > 0 ? 'rounded-l' : '' }}"></div>
+                                <div style="width: {{ $softcoreProgressBarWidth }}%"
+                                        class="bg-[rgb(11,113,193)] h-full {{ $hardcoreProgressBarWidth === 0 ? 'rounded-l' : '' }}"></div>
+                            </div>
+                            </td>
+                        @endif
```

Comment:
This seems like it should be a reusable component.

Reply:
Made the progress bar a component (inner else). the outer two levels of ifs stay here, along with some logic for extracting the values from $userProgress.